tt_eager/tt_dnn/op_library/conv/optimized_conv_op.cpp

```diff
@@ -258,6 +258,200 @@ operation::OpPerformanceModel OptimizedConv::create_op_performance_model(const s
+            auto shard_spec = ShardSpec{shard_grid, shard_shape, ShardOrientation::ROW_MAJOR};
+            auto mem_config = this->output_mem_config;
+            mem_config.shard_spec = shard_spec;
+            return {create_sharded_device_tensor(output_shape, this->output_dtype, output_layout, input_tensor.device(), mem_config)};
+        } else {
+            auto [act_matrix_shape, act_matrix_shape_unpadded] = optimized_conv_op_utils::compute_opt_conv_activation_as_mm_shape(this->input_tensor_shape, conv_params, this->parallelization_config.per_core_out_matrix_height_ntiles, extra_padding_for_32B_alignment);
+            uint32_t act_matrix_height = (uint32_t) act_matrix_shape[1];
+            uint32_t act_matrix_height_ntiles = act_matrix_height / TILE_HEIGHT;
+            uint32_t total_active_num_cores_per_weight_slice = act_matrix_height_ntiles / this->parallelization_config.per_core_out_matrix_height_ntiles;
+            uint32_t weight_matrix_width = weight_tensor.get_legacy_shape()[-1];
+            uint32_t weight_matrix_width_ntiles = weight_matrix_width / TILE_WIDTH;
+            uint32_t num_weight_slices_width = weight_matrix_width_ntiles / this->parallelization_config.per_core_out_matrix_width_ntiles ;
+            uint32_t total_active_num_cores = total_active_num_cores_per_weight_slice * num_weight_slices_width;
+            CoreRangeSet shard_grid = num_cores_to_corerange_set(total_active_num_cores, this->parallelization_config.grid_size, true);
+            std::array<uint32_t, 2> shard_shape = {this->parallelization_config.per_core_out_matrix_height_ntiles * TILE_HEIGHT, this->parallelization_config.per_core_out_matrix_width_ntiles * TILE_WIDTH};
+            auto shard_spec = ShardSpec{shard_grid, shard_shape, this->output_mem_config.shard_spec.value().orientation};
+            auto mem_config = this->output_mem_config;
+            mem_config.shard_spec = shard_spec;
+            return {create_sharded_device_tensor(output_shape, this->output_dtype, output_layout, input_tensor.device(), mem_config)};
+        }
+
+    }
+    return operation::generic_create_output_tensors(*this, input_tensors, this->output_dtype, output_layout, this->output_mem_config);
+}
+
+operation::ProgramWithCallbacks OptimizedConvNew::create_program(const std::vector<Tensor>& input_tensors,
+                                                     const std::vector<std::optional<const Tensor>>& optional_input_tensors,
+                                                     std::vector<Tensor>& output_tensors) const {
+    const auto& input_tensor_a = input_tensors.at(0);
+    const auto& input_tensor_b = input_tensors.at(1);
+    const auto& input_tensor_bias = optional_input_tensors.at(0);
+    auto& output_tensor = output_tensors.at(0);
+    TT_ASSERT(input_tensor_a.memory_config().is_sharded()); // TODO: move this check to validate_input_tensors
```

Comment:
Any validation of user input, use TT_FATAL

TT_ASSERT is for validating internal behaviour (ie allocated Buffer is not null) 
Asserts are disabled in the "release" build for faster speed

TT_FATAL is always on , in all builds ,since we always want to error out if user is not using the OP correctly (even in "release" build)